File: QuasselDroidNG/src/main/java/de/kuschku/libquassel/primitives/serializers/VoidSerializer.java
package de.kuschku.libquassel.primitives.serializers;


import android.support.annotation.Nullable;

import java.io.IOException;
import java.nio.ByteBuffer;
import java.nio.channels.ByteChannel;

public class VoidSerializer implements PrimitiveSerializer<Void> {
    private static final VoidSerializer serializer = new VoidSerializer();

    private VoidSerializer() {
    }

    public static VoidSerializer get() {
        return serializer;
    }

    @Override
    public void serialize(final ByteChannel channel, final Void data) throws IOException {

    }

    @Nullable
    @Override
    public Void deserialize(final ByteBuffer buffer) throws IOException {
        return null;
    }
}

File: lib/src/main/java/de/kuschku/libquassel/protocol/primitive/serializer/VoidSerializer.kt
package de.kuschku.libquassel.protocol.primitive.serializer

import de.kuschku.libquassel.quassel.QuasselFeatures
import de.kuschku.libquassel.util.nio.ChainedByteBuffer
import java.nio.ByteBuffer

object VoidSerializer : Serializer<Any?> {
  override fun serialize(buffer: ChainedByteBuffer, data: Any?, features: QuasselFeatures) {
  }

  override fun deserialize(buffer: ByteBuffer, features: QuasselFeatures): Any? {
    return null
  }
}
